sw/host/opentitanlib/src/transport/cw310/mod.rs

```diff
@@ -231,6 +230,19 @@ impl Transport for CW310 {
             usb.pll_out_enable(2, false)?;
             usb.pll_write_defaults()?;
             Ok(None)
+        } else if let Some(reset_and_clear) = action.downcast_ref::<ResetAndClearUartRx>() {
+            // Assert reset signal (active low) and sleep.
+            // Note: `TransportWrapper::apply_pin_strapping` would be better but this is in
+            // `Transport`.
+            log::info!("Clearing UART RX buffer while holding the target in reset");
+            let reset_pin = self.gpio_pin(Self::PIN_SRST)?;
+            reset_pin.write(false)?;
+            std::thread::sleep(reset_and_clear.reset_duration);
+            // Clear the RX buffer of the console UART.
+            self.uart("0")?.clear_rx_buffer()?;
```

Comment:
LGTM, I think this addresses the concern @a-will has by ensuring we're not in reset when the uart is being cleared. I was trying to do something like this in #13174, and @cfrantz had me drop it, so I think there's still something I don't fully understand, (I was trying to initialize the uart at this point which I don't think had the effect of clearing the buffer, which may have been the concern.)

We definitely don't want to be handling pre-reset messages after commanding something to reset and clear.

Reply:
> LGTM, I think this addresses the concern @a-will has by ensuring we're not in reset when the uart is being cleared.

Might be some wires crossed there, heh. Ideally, we'd clear the buffer while the chip is held in reset. The reset pin would behave like a mutex that protects our UART buffer-clearing critical section ;)

Reply:
Yeah, This is a sense inversion typo, I'll edit the comment.